### ISSUE TITLE
Pretty Printer Oddity with Infix Operators: `a (==) b`

### ISSUE THREAD

**Issue by weaversa** [GaloisInc/cryptol#256]
```
Main> :t [True] == [False]
[True] (==) [False] : Bool
```
What's `Bool`? I thought Cryptol's base type was called `Bit`?

I see `Bit` still works. Can we get rid of `Bool`?

**Comment by weaversa** [comment]
I only see `Bool` in the manual once, as a predefined type synonym.
```
type Bool = Bit
```
Oddly enough, the same page (p.33) gives a different answer than the interpreter for two reasons. `Bool` vs. `Bit` and `(==`) vs `==`.

```
Cryptol> :t (==)
(==) : {a} (Cmp a) => a -> a -> Bool
```

**Comment by weaversa** [comment]
More strangeness...
```
:t \x -> True != x
\x -> True (!=) x : Bit -> Bool
:t \x -> True && x
\x -> True (&&) x : Bit -> Bit
```

**Comment by weaversa** [comment]
Figured it out! The prelude uses `Bool` to define the type of many of the logical and comparison operators. Clearly, those types should use `Bit` instead of `Bool`.

**Comment by TomMD** [comment]
Yes, I was just saying as much.  I'll push a patch but leave this issue open because clearly we should not see:

```
\x -> True (&&) x : Bit -> Bit
```

and instead see:

```
\x -> True && x : Bit -> Bit
```

**Comment by elliottt** [comment]
`(==)` vs `==` is currently just a pretty-printing bug, they're the same operator :)

**Comment by brianhuffman** [comment]
This appears to have been fixed in revision 88e5965e.
